test/workbox-window/integration/test.js

```diff
@@ -0,0 +1,332 @@
+/*
+  Copyright 2019 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+/* global Workbox, sinon */
+/* eslint-disable no-console */
+
+
+const {expect} = require('chai');
+const templateData = require('../../../infra/testing/server/template-data');
+const waitUntil = require('../../../infra/testing/wait-until');
+
+// Store local references of these globals.
+const {webdriver, server} = global.__workbox;
+
+/**
+ * Executes the passed function (and args) async and logs any errors that
+ * occur. Errors are assumed to be passed to the callback as an object
+ * with the `error` property.
+ *
+ * @param {...*} args
+ * @return {*}
+ */
+const executeAsyncAndCatch = async (...args) => {
```

Comment:
Yeah, I was going to, but I also wasn't super happy with it. I wish there was a better way to run `executeAsyncScript()` and have it automatically report errors, but since we can't pass helpers from the page scope in, it's really tricky.

Reply:
I'm fine with the actual code. It was mainly a comment about putting code that might be reused in multiple tests into its own standalone file in our `/infra/` directory.

Is there something about the way the integration test environment is set up that prevents that?

Reply:
Nope, it was just that I wasn't happy with it, so I didn't want to generalize it yet, but I certainly can.

Reply:
Added all the helpers to a new `/infra/testing/webdriver` directory.